### DISCUSSION TITLE
Client not receiving server information

### DISCUSSION THREAD

**Discussion by nitchimon** [projectdiscovery/interactsh#151] [Q&A] [answered]
I have an interesting problem.

I start the Server on a linux system.  On the same system, I start the Client in a different ssh session. I found initially that the Server required  -origin-url flag set to be able to talk to the client.  Once set, it worked.

Then it stopped working.  I did not change any settings.
Why would this occur ?  Work then stop talking?

Is HTTPS a set requirement ?  I am not using HTTPS, only listening onI'm not using a cert.
also for the -domain,  I am using an internal domain  "my.corp.local".  I see it fails to get the cert from letsencrypt.org or create one.

I can see that initially they talk to each other where -debug is set, the server sets the correlationID key.  The Client accepts and then nothing.  No more interactions at all.

For example purposes, I changed the internal IPs and changed the Token.

Server:   
./interactsh-server -domain my.corp.local -listen-ip 10.10.10.5 -debug -token 101929838747 -auth -hostmaster mytest@test.com -origin-html http://10.10.10.5

Client:
interact-client -v -server http://10.10.10.5 -token 101929838747 -o mytest.txt

any thoughts or ideas ?

thanks

**Comment by ehsandeep** [comment, marked as answer]
`origin-url` is only required for the self-hosted web client, so in this case you can avoid using this flag, also for the CLI client, you need to provide a domain URL as input instead of IP, for example:- `interact-client -v -server http://my.corp.local`, assuming you already configured your domain as described in the readme.

**Reply by nitchimon** [reply]
Thank you for the response.
It works again.
thank you
